src/EnergyPlus/ElectricPowerServiceManager.cc

```diff
@@ -3944,6 +3944,8 @@ void ElectricStorage::simulateSimpleBucketModel(EnergyPlusData &state,
                                                 Real64 const controlSOCMaxFracLimit,
                                                 Real64 const controlSOCMinFracLimit)
 {
+    Real64 const minEfficiency = 0.001;
+
```

Comment:
See below--I have followed your suggestion and simplified the code in the process.  There are now separate efficiency parameters for charging and discharging and the actual values are simply overwritten when the efficiencies are less than the minimum.  There is now a single equation for each case rather than 2.  Hopefully the ci comes back clean and it will be ready for re-review soon.

Reply:
So, this will just silently set efficiency to 0.001? Seems like that should either be a severe error (and require a different input) or at least warn that it's been changed.

Reply:
The limits on the input should prevent this from happening, but I will put in a warning message here that the value has been reset.

Reply:
Wait, is this being applied during the simulation? Maybe I'm misunderstanding what's happening here. I thought this was a check on a simple input.

Reply:
Warning messages have now been added for both efficiencies.